lib/prometheus/engineering_collector.rb

```diff
@@ -28,12 +28,14 @@ def type
   end
 
   def dangling(model)
-    model.where.not(host_id: Host.select(:id))
+    model.where.not(host_id: Host.with_policy_or_test_results.select(:id))
   end
 
   def collect
     @dangling_accounts.observe(
-      Account.where.not(account_number: Host.select(:account)).count
+      Account.where.not(
+        account_number: Host.with_policy_or_test_results.select(:account)
```

Comment:
Josef had a suggestion to add a `DISTINCT` here. Maybe it can also speed it up a bit.

Reply:
distinct hosts or accounts?

Reply:
```
Host.with_policy_or_test_results.select(:account).distinct
```

Reply:
Updated